### PR TITLE
Pull request for code updation

### PR DESCRIPTION
**Pull request by manavrmoorthy**
The summary of changes included are as follows:

chapter 2:
- tensorflow 2.0.0 does not run on python 3.10 (version used by colab currently)
- tensorflow gpu removed: 
	https://github.com/tensorflow/tensorflow/releases/tag/v2.12.0
- upgrading tf-explain to 0.3.1 - and explainer.explain arguments ordering has changed, so made that generic

https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/173 - don't know the seriousness of this issue, skipping it for now. i am getting comparable times for training both models when i tried it, and there is nothing that specifically is needed to be changed in the code here for this

chapter 3:
- adding kaggle CLI support to make data download easier - to download any data from kaggle and start training models
- fit_generator -> fit function, and predict_generator -> predict function since they now support generator objects and the former functions are going to get deprecated soon
- the gcloud sessions are not using the same runtime (so we need to re-download data) - using the same strategies to download through kaggle CLI in both notebooks 1 and 2
- deleted deprecated files
- formatted the code

chapter 4:
- removing deprecated files
- the gcloud sessions are not using the same runtime (so we need to re-download data) so writing outputs to gdrive to make sure things are running and outputs are accessible for subsequent notebooks in colab
- renaming 4/1 notebook - since it had underscores instead of hyphens, the colab link in the notebook was not working
- xception added to imports for notebook 1, since it is part of the model_maker
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/163 - fixed
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/164 - fixed
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/169 - fixed
- metric='angular' added for annoy as default arg will be removed in subsequent releases
- removing a duplicate PCA + Annoy section
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/170 - fixed
- time is a negligible factor here, and we do not need it in the plots (since we are using optimised accuracy calculation using numpy from issue 170) - hence, modifying the plots
- removing matplotlib.style.use('seaborn') since it is deprecated
- the final fine-tuning notebook uses Caltech256 features (as per the book), which do not exist, since fine-tuning was done on Caltech101 - hence, renaming those files to caltech101. Can we retain caltech101 to test?
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/167 - fixed, if the above is okay
- formatted the code

chapter 5:
- write_grads and batch_size params have been removed from callback, or will be removed in subsequent releases
- https://github.com/PracticalDL/Practical-Deep-Learning-Book/issues/174 - not able to replicate this issue
- added a pointer to the notebook that suggests that for tensorboard to work without a 403 Forbidden error on Colab, cookies need to be allowed (I faced this issue)
- notebook 3 in chapter 5 is the exact same as notebook 2 in chapter 2 - replaced the file directly
- the autokeras notebook in Colab is named autokeras-error.ipynb - where can we change this to autokeras.ipynb?
- fixing accuracy score calculation in the autokeras notebook
- formatted the code

chapter 6:
- including the download_sample_image function
- formatted the code

chapter 7: no code to update

NOTE - will edit this, as I add more changes.

### CODE REVIEW

File: code/chapter-2/visualization.py [hunk 1]
```diff
@@ -117,60 +128,83 @@ def make_gradcam_heatmap(
...
     superimposed_img = jet_heatmap * 0.4 + img
-    superimposed_img = tensorflow.keras.preprocessing.image.array_to_img(superimposed_img)
+    superimposed_img = tensorflow.keras.preprocessing.image.array_to_img(
+        superimposed_img
+    )
 
-    #Save the the superimposed image to the output path
+    # Save the the superimposed image to the output path
     superimposed_img.save(output_path)
+    print("SAVED")
```
Comment (meherkasam) [inline code/chapter-2/visualization.py:137]: Remove line

File: code/chapter-2/visualization.py [hunk 2]
```diff
@@ -117,60 +128,83 @@ def make_gradcam_heatmap(
...
+    # Creates directory output directoy if it doesn't already exist
     if not os.path.exists(output_dir):
         os.makedirs(output_dir)
     for input_path in sorted(glob.glob(videoframes_path + "/*.jpg")):
         counter += 1
 
-        output_path = output_dir + "/result-" + str(counter).zfill(4) + '.jpg'
+        output_path = output_dir + "/result-" + str(counter).zfill(4) + ".jpg"
+        print(output_path)
+
```
Comment (meherkasam) [inline code/chapter-2/visualization.py:154]: Remove line
